src/jdk.zipfs/share/classes/jdk/nio/zipfs/ZipFileSystem.java

```diff
@@ -2877,7 +2876,7 @@ private int writeCEN(OutputStream os) throws IOException {
                 eoff += (4 + sz);
             }
             if (!foundExtraTime) {
-                if (isWindows) {             // use NTFS
+                if (OperatingSystem.isWindows()) { // use NTFS
                     elenNTFS = 36;           // total 36 bytes
                 } else {                     // Extended Timestamp otherwise
                     elenEXTT = 9;            // only mtime in cen
```

Comment:
It might be better to drop ZipFileSystem from this patch and instead create a bug to re-examine this code. I would have expected it use NFTS when the timestamp is beyond the range of an extended timestamp.

Reply:
Created [JDK-8305662](https://bugs.openjdk.org/browse/JDK-8305662) to track refactoring.
Will revert.

Reply:
> Created [JDK-8305662](https://bugs.openjdk.org/browse/JDK-8305662) to track refactoring.
> Will revert.

Thanks, you can revert the qualified export in module-info and the additional grant in default.policy too.

The interesting thing about this PR is that each change begs the question on whether the code should be refactored, that would take time of course.